compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/FirExpressionsResolveTransformer.kt

```diff
@@ -69,6 +68,7 @@ open class FirExpressionsResolveTransformer(transformer: FirBodyResolveTransform
         qualifiedAccessExpression.transformAnnotations(this, data)
         qualifiedAccessExpression.transformTypeArguments(transformer, ResolutionMode.ContextIndependent)
 
+        val wasField = (qualifiedAccessExpression.calleeReference as? FirNamedReference)?.name == NAME_FOR_BACKING_FIELD
```

Comment:
This can work in 99% cases but looks very hacky. E.g. it can be broken by a local `val field`.

Reply:
I didn't know that `field` isn't a hard keyword: https://kotlinlang.org/docs/reference/keyword-reference.html#soft-keywords 😮  That means, the condition below should also check if the resolved property indeed matches the enclosing property.

We don't have a special construct to wrap backing field `field`; it's just named reference up to this point, and that's why I thought it's inevitable to look at name `field` itself. But, speaking of checking the enclosing property, I just came up with an alternative: once resolved, if that is a property, and if that backing field access is within its own accessors, now we know that property needs the presence of the backing field. Will try this.

Reply:
I have no idea why I have checked the name `field` and backing field reference below together. Checking the backing field reference is simply enough, since it literally means a backing field of a property is referenced! Also, considering where to mark, I found it's better to do something when the backing field reference is generated, not here. Updated the other place instead.